src/Waives.Http/HttpRequestMessageTemplate.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+
+namespace Waives.Http
+{
+    public class HttpRequestMessageTemplate
```

Comment:
✋ This class and all it's properties should be internal.
👏 I really like that you introduced a class for this, rather than passing around a set of arguments each time.